### ISSUE TITLE
spack modulefiles prepend-path --delim 

### ISSUE THREAD

**Issue by luigi-calori** [LLNL/spack#1003]
My system has Modules Tcl 3.1.6 and does not like  prepend-path --delim syntax 
Would it be possible to make these formats configurable in modules .yaml?

Does this problem affect someone else?   I got it by spack load <any newly installed pkg>

**Comment by trws** [comment]
According to their website, Modules TCL added `--delim` in Modules 3.2.3 in 2006.  What distro are you using? Even RedHat EL/CentOS 5 is using 3.2.10 if its up to date.

I'm not averse to having an option for whether that should be supported or not I suppose, but it will limit the functionality of the lua package to not be able to select the correct delimiter for its path variables.

**Comment by luigi-calori** [comment]
The problem is that our sysadms have used this old module installation as a default for all users.
So the default module interpreter for all users is this old one.
I have almost ready a PR  for let the current behaviour be customizable by adding a  delim_format definition in [modules.yaml](https://github.com/RemoteConnectionManager/spack/commit/786d1df0f1e1e20bfc8404455364a6487932e479)

**Comment by luigi-calori** [comment]
Sorry ...  pressed wrong button....
I would like to increase the ability to customize the modules in order for me to be able to comply to our internal policy of module development. 
I'm not yet  fully confident on config syntax but the idea is to let the  user override in modules.yaml the templates that are currently fixed in modules.py

**Comment by trws** [comment]
No worries.  I'm all for having it be more configurable as long as we can express the values we need, I'm sorry the --delim option caused you issues.

**Comment by alalazo** [comment]
@luigi-calori Why not `spack install lmod`  and then use that ?

**Comment by luigi-calori** [comment]
Eh... I have my trouble in trying to show spack is ready to test and to substitute our legacy way of building sw.
I would like to rely on the same module system, this way If I build packages with spack, then I could let other users test them by just doing
module use <spack root>/share/spack/modules/$(spack arch)

They even do not need to be aware modules have been built with spack 

I do not know lmod but I think it will need to be activated and not sure if it can easily interoperate with classical env modules.
I would really like spack adoption to be as less intrusive as possible, especially to users.
I also think that being able to override module templates could be of general utility: for example to add code to log module usage to all modules ( if not already possible)

**Comment by alalazo** [comment]
> I do not know lmod but I think it will need to be activated and not sure if it can easily interoperate with classical env modules

It's completely backward compatible with tcl module files, see lmod docs. 

> I would like to rely on the same module system, this way If I build packages with spack, then I could let other users test them by just doing
module use /share/spack/modules/$(spack arch)

What we have done here to deal with the same kind of issue is to have a short shell script to source lmod init file and add `/share/spack/modules/$(spack arch)` to the `MODULEPATH`. The same script can restore the login setting if a flag is passed.

**Comment by luigi-calori** [comment]
Well, I understand lmod would be the way to go in the long run, but  
I would prefer to not change also the module system and rely on the old one.
As spack allows the production of tcl module code, I thought lmod was not strictly needed.
The --delim was added very recently so I tried to ad a PR that will allow to configure it in order to disable.
Do you think the idea of letting users override module template is dangerous or unconvenient?

**Comment by alalazo** [comment]
> Do you think the idea of letting users override module template is dangerous or unconvenient?

No, I just don't see how much useful would this be in `develop`. Per @trws comment : 

> According to their website, Modules TCL added --delim in Modules 3.2.3 in 2006. What distro are you using? Even RedHat EL/CentOS 5 is using 3.2.10 if its up to date.

we would need this change to support version of environment modules that pre-date 2006. Just to give you an idea of how old this is, think of `gcc-3.4.6`...

**Comment by luigi-calori** [comment]
MM.. I do not know the reason, but on all our clusters we still have module 3.1.6. 
I agree that this is really old and I' ve requested to upgrade.
Nevertheless,  allowing users override  modulefile templates  string defaults  could be useful in other cases. But maybe there are reason for not going in such direction.
If no one else is interested, I' ll keep the PR live until I manage to upgrade my module system.
Maybe you could evaluate the idea of making module code generation more configurable by letting user override  the strings used for template in module.yaml
